src/hotspot/share/gc/epsilon/epsilonHeap.cpp

```diff
@@ -150,15 +152,15 @@ HeapWord* EpsilonHeap::allocate_work(size_t size) {
   size_t used = _space->used();
 
   // Allocation successful, update counters
-  {
+  if (Metaspace::initialized()) {
     size_t last = _last_counter_update;
     if ((used - last >= _step_counter_update) && Atomic::cmpxchg(&_last_counter_update, last, used) == last) {
       _monitoring_support->update_counters();
     }
   }
 
   // ...and print the occupancy line, if needed
-  {
+  if (Metaspace::initialized()) {
```

Comment:
Ditto here. Maybe moving this check to `EpsilonHeap::print_metaspace_info()` and printing a relevant `log_info(gc, metaspace)` is better.